src/referrals-queue/referrals-queue.component.tsx

```diff
@@ -1,35 +1,132 @@
         <div className={styles.greeting}>
           <Trans i18nKey="referrals-queue">Referrals Queue</Trans>
         </div>
+        <div>
+          <div className="omrs-input-group">
+            <label>
+              <Trans i18nKey="from">From</Trans>
+              <div className="omrs-datepicker">
+                <input
+                  type="date"
+                  name="datepicker"
+                  value={fromDate}
+                  onChange={e => setFromDate(e.target.value)}
+                  required
+                />
+                <svg className="omrs-icon" role="img">
+                  <use xlinkHref="#omrs-icon-calendar"></use>
+                </svg>
+              </div>
+            </label>
+          </div>
+          <div className="omrs-input-group">
+            <label>
+              <Trans i18nKey="to">To</Trans>
+              <div className="omrs-datepicker">
+                <input
+                  type="date"
+                  name="datepicker"
+                  value={toDate}
+                  onChange={e => setToDate(e.target.value)}
+                  required
+                />
+                <svg className="omrs-icon" role="img">
+                  <use xlinkHref="#omrs-icon-calendar"></use>
+                </svg>
+              </div>
+            </label>
+          </div>
+          <div className="omrs-input-group">
+            <input
+              id="query-input"
+              type="text"
+              value={ptQuery}
+              onChange={e => setPtQuery(e.target.value)}
+              className="omrs-input-outlined"
+            />
+            <label htmlFor="query-input">
+              <Trans i18nKey="filter-by-patient">Filter by patient</Trans>
+            </label>
+          </div>
+          <div className="omrs-input-group">
+            <label>
+              <Trans i18nKey="referrals-queue">Referral Type</Trans>
+              <select
+                value={referralType}
+                onChange={e => setReferralType(e.target.value)}
+              >
+                <option value="">Any</option>
+                {referralTypes.map(t => (
+                  <option key={t} value={t}>
+                    {t}
+                  </option>
+                ))}
+              </select>
+            </label>
+          </div>
+        </div>
         <div className={styles.tableContainer}>
-          <Table referrals={referrals} />
+          <Table referrals={filteredReferrals} />
         </div>
       </div>
     </div>
   );
 }
 
+function matchQuery(referral: Referral, query: string): boolean {
+  return (
+    !query ||
+    prepareQuery(query).every(regexp =>
+      regexp.test(removeDiacritics(referral.patient_name))
+    ) ||
+    new RegExp(query, "i").test(referral.zl_emr_id)
+  );
+}
+
+function prepareQuery(query: string): RegExp[] {
+  return query.split(/\s/).map(token => {
+    const tokenAscii = removeDiacritics(token);
```

Comment:
Naming convention question: does str.normalize convert to Ascii?

Reply:
Oh, good point, it doesn't actually. `removeDiacritics` produces unicode, but with no diacritics. Will rename.